addons/calendar/static/src/views/calendar_provider_config_form_renderer.js

```diff
@@ -0,0 +1,60 @@
+/** @odoo-module **/
+
+import { FormRenderer } from "@web/views/form/form_renderer";
+import { session } from "@web/session";
+import { useService } from "@web/core/utils/hooks";
+
+
+export class CalendarProviderConfigFormRenderer extends FormRenderer {
+    setup() {
+        super.setup();
+        this.orm = useService('orm');
+        this.rpc = useService('rpc');
+    }
+
+    /**
+     * Activate the external sync for the first time, after installing the
+     * relevant submodule if necessary.
+     *
+     * @private
+     */
+    async onConnect(ev) {
+        ev.preventDefault();
+        ev.stopImmediatePropagation();
+        if (!await this.props.record.save({stayInEdition: true})) {
+            return;  // handled by view
+        }
+        await this.orm.call(
+            this.props.record.resModel,
+            'action_calendar_prepare_external_provider_sync',
+            [this.props.record.resId]
+        )
+        // See google/microsoft_calendar for the origin of this shortened version
+        const provider = this.props.record.data.external_calendar_provider;
+        await this.orm.call(
+            'res.users',
+            `restart_${provider}_synchronization`,
```

Comment:
Don't like dynamic calls like that (makes method hard to read, when changing a method name and grepping you don't find it in code, ...). As anyway we have 2 providers, and we will probably have to go through this code if we support more, can't we have some helper to have provider_name -> route to call / methods to call / ... ?

Reply:
I had a mapping from provider => action and route, (see line 40), but that was more lines for code belonging to other modules.
As I said in a previous comment, we could depart from this implementation originating from calendar, and perform these actions on the backend after installing the right module. (JS method won't be available).

Would you like this for this release?